services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -134,8 +165,16 @@ export const DynamicField = ({ name, label, ...props }: Props) => {
   const hydrationValue = props?.hydrate;
   useEffect(() => {
     if (hydrationValue?.length) {
-      setDisplayValues(hydrationValue);
-      append(hydrationValue);
+      if (
+        hydrationValue.length != displayValues.length &&
+        displayValues.length > hydrationValue.length
+      ) {
+        setDisplayValues(displayValues);
+        append(displayValues);
+      } else {
+        setDisplayValues(hydrationValue);
+        append(hydrationValue);
+      }
```

Comment:
So theres a specific scenario that can happen when the user types in a plan and then clicks Add a row. If we don't check for this here then what'll happen is a new row will be added but the updateReport will trigger a refresh of the form which will load in the new hydrated data. This means it will remove the newly added row you just created and cause the user confusion. 

Basically this check is a quick way to look to see the specific scenario when a user adds a new row following typing in an old and make sure things stay consistant.

Reply:
Wow what an edge case. That makes sense though. I'll add a comment here so future devs know what's up.

Reply:
Did a little simplification here too.
```suggestion
    if (hydrationValue?.length) {
      // guard against autosave refresh error (https://bit.ly/3kiE2eE)
      const displayValuesEntered = displayValues.length > hydrationValue.length;
      const valuesToSet = displayValuesEntered ? displayValues : hydrationValue;
      // set and append values
      setDisplayValues(valuesToSet);
      append(valuesToSet);
    } else {
      appendNewRecord();
    }
```

Reply:
Ty!